### PR TITLE
tt-rss: upgrade to 20210902 + migrate to maria DB 10 + add support for DSM 7 

### PR DESCRIPTION
**Pull request by smaarn**
_Motivation:_  Being able to operate under DSM 7 required Maria DB upgrade
_Linked issues:_  #4524 

Supersedes #4840 

Bump required PHP version to 7.2
Prefer using resource mechanism to inject db
Switch to Maria DB 10 and rely on its automatic Maria DB 5 => Maria DB 10 migration process
Fix initial installation phase which should trigger the schema update

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### Tested scenarii

- [x] Upgrade DSM 6 => DSM 7
- [x] DSM 7 Upgrade as is (e.g. reinstalling the package)
- [x] DSM 7 Installation from scratch

Test packages can be found here: https://github.com/smaarn/spksrc/releases/tag/tt-rss_20210902

### CODE REVIEW

**Comment by ymartin59** [comment]
@smaarn Great job.
I have a "personal" request. I would like to avoid to run MariaDB when PostgreSQL (included in DSM) is available for an application.
Would it be possible to select our database (mariadb or embedded postgresql) in setup wizard? CC @hgy59 @publicarray

**Comment by smaarn** [comment]
That could be an option. There are several things to consider on which feedbacks would be much appreciated:
1. Is there a real officially supported pgsql version I should be considering ?
2. Should I try making the mysql dependency optional ?
3. Should I try performing migration automatically or should this be done only on new installation ?

Adding support for pgsql is soon going to be mandatory anyway since it's only being supported for maintenance and bugfixes.

**Review by ymartin59 — APPROVED** [review]

**Comment by ewoks** [comment]
just a friendly poke for visibility here... As I see changes are approved, but there is no merge yet. Any specific reason for it or we can move forward and add few checkboxes in the table next to tt-rss [here](https://github.com/SynoCommunity/spksrc/issues/4524) ?

**Comment by hgy59** [comment]
@smaarn you need to rebase to  master to fix the `dsmappname` in the INFO file. 
This will bring back the tt-rss link to the DSM Applications fixed with #4993.

File: spk/tt-rss/src/conf/resource
Comment (hgy59) [inline spk/tt-rss/src/conf/resource:39]: can we use apache 2.4 instead of 2.2?
```suggestion
      "backend": 2,
```

Comment (smaarn) [reply spk/tt-rss/src/conf/resource:39]: If we go that path it means having different requirements depending on whether DSM 6 or not (currently it's only depending on the 2.2 version).

If that's an accepted approach I could do that.

Comment (smaarn) [reply spk/tt-rss/src/conf/resource:39]: Wait, actually apache 2.2 is not even listed in the dependencies... That wouldn't be changing anything.

**Comment by smaarn** [comment]
Flagged this PR as draft until I complete testing on the newer version which is
* Using apache 2.4
* Rebased

**Comment by smaarn** [comment]
Latest binaries have been published here https://github.com/smaarn/spksrc/releases/tag/tt-rss_20210902_rc2

**Comment by smaarn** [comment]
> @smaarn Great job.
> I have a "personal" request. I would like to avoid to run MariaDB when PostgreSQL (included in DSM) is available for an application.
> Would it be possible to select our database (mariadb or embedded postgresql) in setup wizard? CC @hgy59 @publicarray

Database selection could totally be possible in setup wizard **but** there is a mechanism which may prove to be awkward is the dependency selection mechanism which is not really working here.

TBH I wonder if, given the complexity, it wouldn't be simplier to 
1. have two packages (since now we're working on a "no-arch" package), one dedicated for mysql and one for pgsql, both replacing the former and conflicting with each other.
2. deprecate the -mysql one (if that's even possible)

I will try to have this done next year if you don't mind and would rather have this taken offside from this PR. WDYT ?

**Comment by smaarn** [comment]
> @smaarn you need to rebase to master to fix the `dsmappname` in the INFO file.
> This will bring back the tt-rss link to the DSM Applications fixed with #4993.

@hgy59 just tested and it's indeed back :)

**Comment by smaarn** [comment]
Out of curiosity, is there anything specific blocking this PR ? If yes, is there anything I can do to help ?

The postgresql project is indeed a topic but shouldn't it be considered separately ?

**Comment by smaarn** [comment]
Rebased at head

**Comment by smaarn** [comment]
Had to rebase at head since there were conflicts.

**Comment by smaarn** [comment]
Rebased @ head to make sure to keep up with changes on the master branch.

**Comment by smaarn** [comment]
Rebased @ head to handle conflict in merge.

Any objection to my merging ? (Knowing that obviously "just merging" won't be sufficient but at least it's a first towards releasing :) )

I am currently working on making the postgresql adaptation but I don't believe it would be relevant to mix it with this MR (which already changes a few things...)

File: spk/tt-rss/src/service-setup.sh
Comment (hgy59) [inline spk/tt-rss/src/service-setup.sh:1]: this file is included into other scripts files and shebang is not required.
```suggestion
```

**Comment by smaarn** [comment]
Rebased at head before merge by EOD (CET time).

**Comment by hgy59** [comment]
@smaarn is this ready to be published?
_(sorry I was out for vacation from mid september to mid october)_

**Comment by smaarn** [comment]
@hgy59 yed it should be ready to publish. Thanks a lot !

**Comment by hgy59** [comment]
@smaarn to create the noarch package for SRM 1.1 I had to temporary deactivate the `REQUIRED_MIN_DSM = 6.0`

**Pull request merged**
